tests/org.dataflowanalysis.converter.tests/src/org/dataflowanalysis/converter/tests/PCMTest.java

```diff
@@ -87,68 +92,59 @@ private void testSpecificModel(String inputModel, String inputFile, String model
 
         var dfd = complete.dataFlowDiagram();
         var dd = complete.dataDictionary();
+        
 
-        assertEquals(dfd.getNodes()
-                .size(),
-                assIdToName.keySet()
-                        .size());
-
-        List<String> nodeIds = new ArrayList<>();
-        for (Node node : dfd.getNodes()) {
-            nodeIds.add(node.getId());
-        }
-        Collections.sort(nodeIds);
-        List<String> assIds = new ArrayList<>(assIdToName.keySet());
-        Collections.sort(assIds);
-
-        assertEquals(assIds, nodeIds);
-
-        for (Node node : dfd.getNodes()) {
-            assertEquals(node.getEntityName(), assIdToName.get(node.getId()));
-        }
-
-        Set<String> flowNames = new HashSet<>();
-        for (AbstractTransposeFlowGraph abstractTransposeFlowGraph : flowGraph.getTransposeFlowGraphs()) {
-            AbstractVertex<?> previousAse = null;
-            for (AbstractVertex<?> ase : abstractTransposeFlowGraph.getVertices()) {
-                if(!ase.getPreviousElements().isEmpty()) {
-                    List<DataCharacteristic> variables = ase.getAllDataCharacteristics();
-                    if (variables.isEmpty()) flowNames.add(previousAse.hashCode()+ase.hashCode()+"");
-                    for (DataCharacteristic variable : variables) {
-                        flowNames.add(previousAse.hashCode()+ase.hashCode()+variable.variableName());
-                    } 
-                }
-                previousAse=ase;
-            }
-        }
-
-        assertEquals(flowNames.size(), dfd.getFlows()
```

Comment:
Please use the pcm tfg and the dfd tfg to check if we have the same amount of flows

Reply:
Implemented in 6acdf117b40c